From 4805b0c5dfc838145d8e8d81aaf5f0d9b7ab9114 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Mon, 15 May 2023 15:40:35 +0800
Subject: [PATCH 240/263] [Huawei] Revert "[Backport] filemap: Correct the
 conditions for marking a folio as accessed"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186896

--------------------------------

This reverts commit 348e887e6975c357ed0fbca352564f817a377bca.

Because this commit make a mistake to judge if the page is the same.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 mm/filemap.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/mm/filemap.c b/mm/filemap.c
index 573d3d1af3e6..7e79f3a9c5dd 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -2530,13 +2530,6 @@ static int generic_file_buffered_read_get_pages(struct kiocb *iocb,
 	goto find_page;
 }
 
-static inline bool pos_same_page(loff_t pos1, loff_t pos2, struct page *page)
-{
-	unsigned int shift = page_shift(page);
-
-	return (pos1 >> shift == pos2 >> shift);
-}
-
 /**
  * generic_file_buffered_read - generic file read routine
  * @iocb:	the iocb to read
@@ -2627,10 +2620,11 @@ ssize_t generic_file_buffered_read(struct kiocb *iocb,
 		writably_mapped = mapping_writably_mapped(mapping);
 
 		/*
-		 * When a read accesses a page several times, only
+		 * When a sequential read accesses a page several times, only
 		 * mark it as accessed the first time.
 		 */
-		if (pos_same_page(iocb->ki_pos, ra->prev_pos -1, pages[0]))
+		if (iocb->ki_pos >> PAGE_SHIFT !=
+		    ra->prev_pos >> PAGE_SHIFT)
 			mark_page_accessed(pages[0]);
 		for (i = 1; i < pg_nr; i++)
 			mark_page_accessed(pages[i]);
-- 
2.17.1

